### ISSUE TITLE
R Crashes when generating plots with large amounts of data

### ISSUE THREAD

**Issue by aforsythe** [rocker-org/rocker#389]
Sorry for the poorly formed issue.

I'm currently using the tidyverse docker container.  I have an issue where R is crashing when generating a ggplot using a large amounts of data ... but only on some hosts.  I'm trying to figure out a way to debug but logs don't seem to exist in the normal places. 

Any thoughts would be greatly appreciated.

**Comment by cboettig** [comment]
Check the logs of the host machine, or simply keep an eye on resource use of the host machine.  Sounds like you are probably running out of memory.  As discussed in other threads, the open source RStudio Server doesn't log much

**Comment by eddelbuettel** [comment]
(Also, if I may: wrong repo.  The version stack lives here: https://github.com/rocker-org/rocker-versioned and you could always check issues there as well.)

Without a [MCVE](https://stackoverflow.com/help/mcve) it will be hard to help.  And once you have a MCVE, you can also help yourself:  does it happen on non-Rocker? Yes? No? Does it depend on resources?  If so, which?

As you can imagine, `ggplot()` is fairly widely used and tested, including on Rocker.  So unless we have some more 'meat' here we may have to close this.

**Comment by aforsythe** [comment]
@cboettig memory doesn't seem to be the problem.  That was the first thing I thought of but a Macbook Pro with 32GB of RAM dedicated to docker has no problem while an iMac Pro with 96GB RAM dedicated to docker crashes.  Either seems to approach memory limits when running the `ggplot`.

@eddelbuettel Thanks for the link to the other repo and totally understand where you're coming from on the MCVE.  Obviously I have a Dockerfile but I'm trying to figure out why the issue isn't reproducible across hosts.

**Comment by eddelbuettel** [comment]
Could be anything. Change graphics devices: change you plot to / save to `pdf()`?  `png()`?  ...

**Comment by aforsythe** [comment]
@eddelbuettel thanks for the hint ... so a little more info.  The exact function I'm using isn't generic `ggplot` rather [`VlnPlot`](https://www.rdocumentation.org/packages/Seurat/versions/3.1.4/topics/VlnPlot) from [`Seurat`](https://satijalab.org/seurat/).  

doing 
```
p<-VlnPlot(...)
pdf('test.pdf') or png('test.png')
print(p)
```
crashes at the `print(p)` line producing a `test.png` or `test.pdf` of zero bytes.
Again works on one host but not another.

**Comment by cboettig** [comment]
@aforsythe if you are getting different behaviour on the different hosts, this really suggests the difference is due to differences in the docker setup (consider `docker info`) or the host itself, neither are things we have any control over or way to inspect at our end.

**Comment by eddelbuettel** [comment]
Possibly your user error -- one is supposed to call `dev.off()` to close a device.  

Please tests code on a known working system first.  I concur with Carl that is unlikely to be our beef, especially as we nothing to graphics drivers or device.  So please allow me to close this.  By allo means, feel free to reopen if you have something reproducible involving Rocker.

**Comment by aforsythe** [comment]
of course ... just didn't include `dev.off()` in the example above because R crashes before that command.  Understand why you're closing just looking for guidance.

**Comment by eddelbuettel** [comment]
Ah ok -- good to know.

`Seurat` is also a bit of a large package. Maybe start by drilling down with its code where/why things go sour to narrow it down.

**Comment by nfancy** [comment]
@aforsythe Hi, I am having a similar issue too. When I am trying to read a large sparse matrix on a rstudio container, it crashes. Did you find a solution? Many thanks.

**Comment by docsteveharris** [comment]
FWIW I've had a similar issue that was fixed by downgrading from rocker/tidyverse:4.0.0 to 3.6.3.
I eventually got an error message that when I tested the devel version that indicated that R graphics 14 were not supported with that version of R studio. On 4.0.0 it just crashed with an 'unexpectedly terminated' message.
